src/modules/FtpServer.rb

```diff
@@ -443,10 +443,11 @@ def WriteFirewallSettings
       return true if !firewalld.running?
 
       if Ops.get(@EDIT_SETTINGS, "PassiveMode") == "YES"
-        port_range = Ops.add(
-          Ops.add(Ops.get(@EDIT_SETTINGS, "PasMinPort"), ":"),
-          Ops.get(@EDIT_SETTINGS, "PasMaxPort")
-        )
+        if !valid_port_range(@EDIT_SETTINGS["PasMinPort"], @EDIT_SETTINGS["PasMaxPort"])
```

Comment:
The method expect `String` parameters at least based on documentation, so or adapt the documentation or you could use `@EDIT_SETTINGS.fetch("PasMinPort","")` omitting the nil? check

Reply:
ok, you forced me to go through sources ;-)

Everything interesting is done around here https://github.com/yast/yast-ftp-server/blob/master/src/include/ftp-server/wid_functions.rb#L904

```PasMinPort``` and ```PasMaxPort``` are stored as strings but initialized and validated as integers. So, I'll handle that as integers and change my patch accordingly.

Reply:
But that just validate the entries for the widget, have you really tested that the attributes are Integers at the end?

https://github.com/yast/yast-ftp-server/blob/master/src/include/ftp-server/wid_functions.rb#L922

Reply:
yes, bcs it otherwise will not pass through ```ValidPasMaxPort```

Reply:
Really? 

```ruby
FtpServer.WriteToEditMap(
        "PasMaxPort",
        Builtins.tostring(UI.QueryWidget(Id("PasMaxPort"), :Value))
)
```
Then I'm lost.